web/jobs/jobs.go

```diff
@@ -108,10 +131,72 @@ func pushJob(c echo.Context) error {
 	return nil
 }
 
+func newTrigger(c echo.Context) error {
+	instance := middlewares.GetInstance(c)
+	scheduler := instance.JobsScheduler()
+	req := &apiTriggerRequest{}
+	if err := c.Bind(&req); err != nil {
+		return wrapJobsError(err)
+	}
+
+	t, err := jobs.NewTrigger(&jobs.TriggerInfos{
+		Type:       req.Type,
+		WorkerType: req.WorkerType,
+		Arguments:  req.Arguments,
+		Options:    req.Options,
+		Message: &jobs.Message{
+			Type: jobs.JSONEncoding,
+			Data: req.WorkerArguments,
+		},
+	})
+	if err != nil {
+		return wrapJobsError(err)
+	}
+	if err = scheduler.Add(t); err != nil {
+		return wrapJobsError(err)
+	}
+	return jsonapi.Data(c, http.StatusCreated, &apiTrigger{t}, nil)
+}
+
+func getTrigger(c echo.Context) error {
+	instance := middlewares.GetInstance(c)
+	scheduler := instance.JobsScheduler()
+	t, err := scheduler.Get(c.Param("trigger-id"))
+	if err != nil {
+		return wrapJobsError(err)
+	}
+	return jsonapi.Data(c, http.StatusOK, &apiTrigger{t}, nil)
+}
+
+func deleteTrigger(c echo.Context) error {
+	instance := middlewares.GetInstance(c)
+	scheduler := instance.JobsScheduler()
+	return wrapJobsError(scheduler.Delete(c.Param("trigger-id")))
+}
+
+func getAllTriggers(c echo.Context) error {
+	instance := middlewares.GetInstance(c)
+	scheduler := instance.JobsScheduler()
+	ts, err := scheduler.GetAll()
+	if err != nil {
+		return wrapJobsError(err)
+	}
+	objs := make([]jsonapi.Object, 0, len(ts))
+	for _, t := range ts {
+		objs = append(objs, &apiTrigger{t})
+	}
```

Comment:
I found this trick recently. I don't think there is a difference. However I can fallback on the basic solution for coherency with the rest of the code.